### PR TITLE
Fix ClassNotFoundException problem after #1470

### PR DESCRIPTION
**Pull request by tasomaniac**
## Problem

After #1470 was merged, non-Android pure Kotlin projects would not download/pull Android plugin anymore. This causes `ClassNotFoundException` on `BaseVariant` class from Android. 

## Solution

Split task creation for Android and Jvm only projects. `TaskConfigurator` is introduced to capture the common functionality. Yes, used inheritance so far 🤦‍♂ Groovy does not make it easy.

Android and Jvm subclasses encapsulates fully the task configuration for 2 different platforms.

**Note:** I verified that this works with @martinbonnin 's project: https://github.com/martinbonnin/testAndroidCompileOnly But it would be good if somebody else also does it.

### CODE REVIEW

File: apollo-gradle-plugin/build.gradle [hunk 1]
```diff
@@ -15,6 +15,7 @@ dependencies {
 
   implementation localGroovy()
   implementation project(':apollo-compiler')
+  implementation dep.guavaJre
```
Comment (tasomaniac) [inline apollo-gradle-plugin/build.gradle]: Android was pulling guava too and we were using it. Now, it is explicitly defined.

**Comment by sav007** [comment]
It seems plugin tests don't like changes.

**Comment by martinbonnin** [comment]
`ApolloPlugin.APOLLO_CODEGEN_GENERATE_TASK_NAME` became `TaskConfigurator.APOLLO_CODEGEN_GENERATE_TASK_NAME`.

**Comment by tasomaniac** [comment]
Sorry folks, I fell asleep while waiting for the CI. It was late 😛

**Pull request merged**